### PR TITLE
Cast image dimensions from shortcode as int

### LINKED ISSUE #434: Valid HTML height/width values cause Exception in [image] shortcode
After running the `TagsToShortcode` task, a valid HTML image size of 50% was copied into the tag attribute. 
```html
<img class="image" src="some-image.png" alt="..." width="50%" height="auto" data-id="87299" data-shortcode="image" />
```

This results in an exception in the `ImageShortcodeProvider`: `InvalidArgumentException: Width must be a numeric value`.

While this is probably an edge case, I do think (editor supplied) shortcode values should be sanitized and/or fallback to sane defaults by the handler so they're unable to cause an exception?

# Pull request
- [ ] https://github.com/silverstripe/silverstripe-assets/pull/436

### PR DESCRIPTION
**Pull request by micschk**
This prevents sizes like '50%' from triggering exceptions.
Fixes #434 (or at least prevents an exception)
Blocked by https://github.com/silverstripe/silverstripe-framework/pull/9872

### CODE REVIEW

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
That's an awesome contribution. Would you mind adding a test to `tests/php/Shortcodes/ImageShortcodeProviderTest.php` to cover this edge case?

File: tests/php/Shortcodes/ImageShortcodeProviderTest.php [hunk 1]
```diff
@@ -89,6 +89,24 @@ public function testShortcodeHandlerAddsDefaultAttributes()
...
+        $this->assertEquals(
+            sprintf(
+                '<img src="%s" alt="" width="50%" height="auto">',
+                $image->Link()
+            ),
+            $parser->parse(sprintf(
+                '[image id="%d" alt="" width="50%" height="auto"]',
+                $image->ID
+            ))
+        );
```
Comment (maxime-rainville) [inline tests/php/Shortcodes/ImageShortcodeProviderTest.php:108]: Test is failing because `sprintf` thinks the `%` in `50%` is meant to be another substitution. This should take care of that problem.

```suggestion
        $this->assertEquals(
            sprintf(
                '<img src="%s" alt="" width="50%%" height="auto">',
                $image->Link()
            ),
            $parser->parse(sprintf(
                '[image id="%d" alt="" width="50%%" height="auto"]',
                $image->ID
            ))
        );
```

Test is till failing because we end up with `alt="&quot;&quot;"` in the alt attribute, which must be related to that other issue you raise https://github.com/silverstripe/silverstripe-assets/issues/435

I'll see if I can figure out what's wrong with that one.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: tests/php/Shortcodes/ImageShortcodeProviderTest.php (hunk 1 above)
Comment (maxime-rainville) [reply tests/php/Shortcodes/ImageShortcodeProviderTest.php:108]: Once the above suggestion is applied and https://github.com/silverstripe/silverstripe-framework/pull/9872 is merged, this PR should be green.

**Review by maxime-rainville — APPROVED** [review]
Blocking PR has been merged and I've committed my tweaks to the unit test which are now green.
